Automattic/newspack-plugin
# fix: deactivate Salesforce syncing if RR platform is not Newspack

### All Submissions:






### Changes proposed in this Pull Request:



Fixes an issue where if you had previously set up a Salesforce integration on the Newspack Reader Revenue platform, then changed to a different platform that still uses WooCommerce (e.g. Stripe), the Salesforce webhooks would still fire and attempt to sync, but fail because the orders don't contain the expected data. This fixes the issue by deactivating the webhooks if the wrong Reader Revenue platform is active, which should help clean up the debug log.

### How to test the changes in this Pull Request:

1. On `master`, make sure WooCommerce and all required extensions are installed and active. Set your Reader Revenue platform to "Newspack" and set up a [Salesforce integration](https://newspack.pub/support/reader-revenue/salesforce/). Ping me if you need test credentials. 
2. Change your platform to Stripe and complete Stripe configuration. Make sure to create the required webhooks.
3. On the front-end, make a donation via Stripe.
4. After a few minutes, observe fatals in your debug.log indicating a failed sync. In the WP dashboard, go to **WooCommerce > Orders** and find the order corresponding to your Stripe donation. Observe a "Salesforce sync status" message notice in the Order Actions panel indicating a failed sync.
5. Check out this branch, repeat step 3. This time, confirm there are no PHP errors thrown after the WC order is created, and confirm that the "Salesforce sync status" messaging is no longer shown in WC order pages.

### Other information:
